src/PSRule.Rules.Azure/rules/Azure.MariaDB.Rule.ps1

```diff
@@ -35,6 +35,65 @@ Rule 'Azure.MariaDB.UseSSL' -Ref 'AZR-000334' -Type 'Microsoft.DBforMariaDB/serv
     $Assert.HasFieldValue($TargetObject, 'properties.sslEnforcement', 'Enabled').Reason($LocalizedData.MariaDBEncryptedConnection)
 }
 
+# Synopsis: Azure Database for MariaDB servers should meet naming requirements.
+Rule 'Azure.MariaDB.ServerName' -Ref 'AZR-000336' -Type 'Microsoft.DBforMariaDB/servers' -Tag @{ release = 'GA'; ruleSet = '2022_12'; } {
+    # https://learn.microsoft.com/nb-no/azure/azure-resource-manager/management/resource-name-rules#microsoftdbformariadb
+
+    # Between 3 and 63 characters long
+    $Assert.GreaterOrEqual($PSRule, 'TargetName', 3)
+    $Assert.LessOrEqual($PSRule, 'TargetName', 63)
+
+    # Lowercase letters, numbers, and hyphens
+    # Can't start or end with a hyphen
+    $Assert.Match($PSRule, 'TargetName', '^[a-z0-9]([a-z0-9-]*[a-z0-9]){2,62}$', $True)
+}
+
+# Synopsis: Azure Database for MariaDB databases should meet naming requirements.
+Rule 'Azure.MariaDB.DatabaseName' -Ref 'AZR-000337' -Type 'Microsoft.DBforMariaDB/servers', 'Microsoft.DBforMariaDB/servers/databases' -Tag @{ release = 'GA'; ruleSet = '2022_12'; } {
+    # https://learn.microsoft.com/nb-no/azure/azure-resource-manager/management/resource-name-rules#microsoftdbformariadb
+
+    $databases = @(GetMariaDBDatabaseName)
```

Comment:
This would fail for a server that has no databases.

```suggestion
    $databases = @(GetMariaDBDatabaseName)
    if ($databases.Length -eq 0) {
        return $Assert.Pass()
    }
```